File: Open3D/06_features_fpfh/cpp/main.cpp
#include <open3d/Open3D.h>
#include <iostream>
int main(){ auto pc=open3d::io::CreatePointCloudFromFile("input.ply"); if(!pc) return 1; auto down=pc->VoxelDownSample(0.03); down->EstimateNormals(open3d::geometry::KDTreeSearchParamHybrid(0.06,30)); auto f=open3d::pipelines::registration::ComputeFPFHFeature(*down,open3d::geometry::KDTreeSearchParamHybrid(0.15,100)); std::cout<<f->data_.rows()<<"x"<<f->data_.cols()<<"\n"; }

File: Open3D/06_features_fpfh/python/main.py
import open3d as o3d
pc=o3d.io.read_point_cloud("input.ply").voxel_down_sample(0.03); pc.estimate_normals(o3d.geometry.KDTreeSearchParamHybrid(radius=0.06,max_nn=30)); f=o3d.pipelines.registration.compute_fpfh_feature(pc,o3d.geometry.KDTreeSearchParamHybrid(radius=0.15,max_nn=100)); print(f.data.shape)
